filter/filter_impl/auth/provider_auth.go

```diff
@@ -0,0 +1,39 @@
+package auth
+
+import (
+	"github.com/apache/dubbo-go/common/constant"
+	"github.com/apache/dubbo-go/common/extension"
+	"github.com/apache/dubbo-go/common/logger"
+	"github.com/apache/dubbo-go/filter"
+	"github.com/apache/dubbo-go/protocol"
+)
+
+type ProviderAuthFilter struct {
+}
+
+func init() {
+	extension.SetFilter(constant.PROVIDER_AUTH_FILTER, getProviderAuthFilter)
+}
+
+func (filter *ProviderAuthFilter) Invoke(invoker protocol.Invoker, invocation protocol.Invocation) protocol.Result {
+	logger.Infof("invoking providerAuth filter.")
+	url := invoker.GetUrl()
+	shouldAuth := url.GetParamBool(constant.SERVICE_AUTH_KEY, false)
+	if shouldAuth {
```

Comment:
it seems the same logic to ConsumerSignFilter

Reply:
I don't think this is necessary. The consumer side calls the sign method, but the provider side calls the auth method, and the error handling is not the same. This is an overwrap and does not make the code more readable. Or do you have a better way?